app/gui/src/project-view/composables/nodeColors.ts

```diff
@@ -35,18 +36,18 @@ export function useNodeColors(graphStore: GraphStore, getCssValue: (variable: st
   return { getNodeColor, getNodeColors }
 }
 
-/** TODO: Add docs */
+/** Compute node color based on the node type, group, and type name. */
 export function computeNodeColor(
   getType: () => NodeType,
   getGroup: () => GroupInfo | undefined,
-  getTypeName: () => string | undefined,
+  getTypeName: () => ProjectPath | undefined,
 ) {
   if (getType() === 'output') return 'var(--output-node-color)'
   if (getType() === 'input') return 'var(--output-node-color)'
   const group = getGroup()
   if (group) return groupColorStyle(group)
   const typeName = getTypeName()
-  if (typeName) return colorFromString(typeName)
+  if (typeName) return colorFromString(typeName.path ?? 'Unknown')
```

Comment:
This should include the `project` so that similarly-named types don't end up similarly-coloured too. Why not `.key()`?